components/wso2is.key.manager.tokenpersistence/src/main/java/org/wso2/is/key/manager/tokenpersistence/dao/DBInvalidTokenPersistence.java

```diff
@@ -0,0 +1,400 @@
+            throw new IdentityOAuth2Exception("Error while checking existence of internally revoked JWT for subject "
+                    + "entity id: " + entityId, e);
+        }
+    }
+
+    @Override
+    public void revokeTokensByUserEvent(String subjectId, String subjectIdType,
+                                        long revocationTime, String organization, int retryAttemptCounter)
+            throws IdentityOAuth2Exception {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            String updateQuery = SQLQueries.UPDATE_SUBJECT_ENTITY_REVOKED_EVENT;
+            try (PreparedStatement ps = connection.prepareStatement(updateQuery)) {
+                ps.setTimestamp(1, new Timestamp(revocationTime),
+                        Calendar.getInstance(TimeZone.getTimeZone(PersistenceConstants.UTC)));
+                ps.setString(2, subjectId);
+                ps.setString(3, subjectIdType);
+                ps.setString(4, organization);
+                int rowsAffected = ps.executeUpdate();
+                if (rowsAffected == 0) {
+                    log.debug("User event token revocation rule not found. Inserting new rule.");
+                    IdentityDatabaseUtil.rollbackTransaction(connection);
+                    String insertQuery = SQLQueries.INSERT_SUBJECT_ENTITY_REVOKED_EVENT;
+                    String errorMessage = String.format("User event token revocation rule for subject id : %s, "
+                                    + "type : %s and organization : %s already exists", subjectId,
+                            subjectIdType, organization);
+                    try (PreparedStatement ps1 = connection.prepareStatement(insertQuery)) {
+                        ps1.setString(1, UUID.randomUUID().toString());
+                        ps1.setString(2, subjectId);
+                        ps1.setString(3, subjectIdType);
+                        ps1.setTimestamp(4, new Timestamp(revocationTime),
+                                Calendar.getInstance(TimeZone.getTimeZone(PersistenceConstants.UTC)));
+                        ps1.setString(5, organization);
+                        ps1.execute();
+                        IdentityDatabaseUtil.commitTransaction(connection);
+                        if (retryAttemptCounter > 0) {
+                            log.info("Successfully recovered CON_SUB_EVT_KEY constraint violation with the attempt : "
+                                    + retryAttemptCounter);
+                        }
+                    } catch (SQLIntegrityConstraintViolationException e) {
+                        log.warn("User event token revocation rule already persisted.");
+                        IdentityDatabaseUtil.rollbackTransaction(connection);
+                        if (retryAttemptCounter >= getTokenPersistRetryCount()) {
+                            log.error("CON_SUB_EVT_KEY constraint violation retry count exceeds above the maximum "
```

Comment:
This should be changed to: "CON_SUB_EVT_KEY constraint violation retry count exceeds the maximum". No need to add the word above.